common/main.go

```diff
@@ -0,0 +1,66 @@
+package common
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Log contains the default logger to use.
+var Log = logrus.WithFields(logrus.Fields{
+	"service": "app-exposer",
+	"art-id":  "app-exposer",
+	"group":   "org.cyverse",
+})
+
+// ErrorResponse represents an HTTP response body containing error information. This type implements
+// the error interface so that it can be returned as an error from from existing functions.
+type ErrorResponse struct {
+	Message   string                  `json:"message"`
+	ErrorCode string                  `json:"error_code,omitempty"`
+	Details   *map[string]interface{} `json:"details,omitempty"`
+}
```

Comment:
I know we've been considering switching to `camelCase` member keys in JSON response bodies. I wanted to use `snake_case` here for the error code, however, because that's what our Clojure services use. These response bodies will ultimately be returned to the UI from the apps service, so I thought it would be good to keep the error response bodies fairly consistent.

Our Clojure services have the ability to include top-level custom elements in the response body. It doesn't seem like that would be very easy to do in Go, so I decided to go with a single `details` member that contains all of the custom fields. I don't think this will cause any problems, but I'd be interested to know if anyone can think of some pitfalls that I might have missed.